bedrock/base/middleware.py

```diff
@@ -75,3 +75,28 @@ def process_request(self, request):
         request.path_info = '/' + prefixer.shortened_path
         request.locale = prefixer.locale
         tower.activate(prefixer.locale)
+
+
+class BasicAuthMiddleware(object):
+    """
+    Middleware to protect the entire site with a single basic-auth username and password.
+    Set the BASIC_AUTH_CREDS environment variable to enable.
+    """
+    def process_request(self, request):
+        required_auth = settings.BASIC_AUTH_CREDS
+        if required_auth:
+            if 'HTTP_AUTHORIZATION' in request.META:
+                auth = request.META['HTTP_AUTHORIZATION'].split()
+                if len(auth) == 2:
+                    if auth[0].lower() == "basic":
+                        provided_auth = base64.b64decode(auth[1])
+                        if provided_auth == required_auth:
+                            # we're good. continue on.
+                            return None
+
+        response = HttpResponse(status=401,
```

Comment:
No, but that's a good catch. It should be indented to be inside the topmost if block. It would be fine because of how we're handling the middleware enablement, but it's better to be correct here for sure.